changes.go

```diff
@@ -162,27 +162,44 @@ func (g *Commands) resolveChangeListRecv(
 			return
 		}
 	}
-
-	// TODO: limit the number of active tasks for children lookups
 	dirlist := merge(remoteChildren, localChildren)
+
+	// Arbitrary value. TODO: Calibrate or calculate this value
+	chunkSize := 100
+	srcLen := len(dirlist)
+	chunkCount, remainder := srcLen/chunkSize, srcLen%chunkSize
+	i := 0
+
+	if remainder != 0 {
+		chunkCount += 1
+	}
+
 	var wg sync.WaitGroup
-	wg.Add(len(dirlist))
-	for _, l := range dirlist {
-		go func(wg *sync.WaitGroup, isPush bool, cl *[]*Change, p string, l *dirList) {
+	wg.Add(chunkCount)
+	for j := 0; j < chunkCount; j += 1 {
+		end := i + chunkSize
+		if end >= srcLen {
+			end = srcLen
+		}
+
+		go func(wg *sync.WaitGroup, isPush bool, cl *[]*Change, p string, dlist []*dirList) {
```

Comment:
@ikwyl6 this is how you create co-routines. We are doing it. Just limiting the number of active routines which was the main cause of the problem in the first place.